### PR TITLE
Convert irlink script to python

### LINKED ISSUE #36: src/tools/irlink shouldn't be bash, but python

### PR DESCRIPTION
**Pull request by nbyavuz**
Fixes #36.

### CODE REVIEW

**Comment by anarazel** [comment]
Seems to not build right now?

**Comment by nbyavuz** [comment]
> Seems to not build right now?

The cause is: https://github.com/anarazel/postgres/blob/30f34f8f6f1dd29b300469afcbba89d619b72b22/meson.build#L334

It seems that when we add cpp language, it can't find libxml dependency. I am currently trying to fix that.

File: meson_options.txt [hunk 1]
```diff
@@ -65,7 +65,7 @@ option('libxml', type : 'feature', value: 'auto',
 option('libxslt', type : 'feature', value: 'auto',
   description: 'XSLT support in contrib/xml2')
 
-option('llvm', type : 'feature', value: 'disabled',
+option('llvm', type : 'feature', value: 'auto',
```
Comment (anarazel) [inline meson_options.txt:68]: Why did you change this as part of this PR?

Comment (nbyavuz) [reply meson_options.txt:68]: Since irlink is called in llvm: https://github.com/anarazel/postgres/blob/30f34f8f6f1dd29b300469afcbba89d619b72b22/meson.build#L375 
I changed that for running irlink at meson build and put in seperate commit.

Comment (anarazel) [reply meson_options.txt:68]: I still don't think we should enable llvm by default. For one there's still a few incompatibilities (as you see in the test). But also more generally, it makes the build a *lot* more expensive. So we shouldn't enable it by default.  How about instead just enabling it explicitly for one of the tasks in .cirrus.yml?

**Comment by anarazel** [comment]
Looks good!

**Comment by anarazel** [comment]
Looks good!
